test/basic.ml

```diff
@@ -10,10 +10,61 @@ module V = struct
   type envelope = Rtree.Rectangle.t
 
   let envelope i = List.assoc i !pre_made_envelopes
+
+  let mindist _p _e = 0.
+  let minmaxdist _p _e = 0.
 end
 
 module R = Rtree.Make (Rtree.Rectangle) (V)
 
+module Vp = struct
+  type t = floatarray
+
+  let t =
+    let to_array arr = Float.Array.to_list arr |> Array.of_list in
+    let of_array arr = Array.to_list arr |> Float.Array.of_list in
+    Repr.(map (array float) of_array to_array)
+  
+
+  type envelope = Rtree.Rectangle.t
+
+  let envelope fa=
+    let x0 =  Array.Floatarray.get fa 0 in
+    let x1 =  Array.Floatarray.get fa 0 in
+    let y0 =  Array.Floatarray.get fa 1 in
+    let y1 =  Array.Floatarray.get fa 1 in
+    Rtree.Rectangle.v ~x0 ~y0 ~x1 ~y1
+
+  let get arr i = Array.Floatarray.get arr i 
+  
+  let mindist (p:floatarray) (e:envelope) = 
+    let (x0,y0,x1,y1) = Rtree.Rectangle.coords e in 
+    let gete ind = if ind==0 then x0
+                else if ind==1 then y0 else if ind==2 then x1 else y1 in
```

Comment:
multiple if-else is not a good idea IMO
Maybe you can use a match statement here, something like this
```
let gete = match ind with 
...
| 1 -> ...
|2 -> ...
...
```

Reply:
Agree with @AryanGodara here, also note the difference between `=` and `==` in OCaml (both can be used to check for _equality_): https://cs3110.github.io/textbook/chapters/mut/refs.html?highlight=physical#physical-equality